### ISSUE TITLE
Add SOCKS5 proxy support from the really beginning (start) of the application

### ISSUE THREAD

**Issue by ghsrcgh** [Aseman-Land/Cutegram#287]
Please add SOCKS5 proxy support from the really beginning (start) of the application. In some countries like China or Iran (and some others, including Russina in the possible future) Telegram servers are blocked. You will unable to connect or register account without SOCKS5 proxy.

**Moreover, official Telegram Desktop client (and now - Mobile client) supporting SOCKS5 proxy from the real beginning of application start.**

Please add this feature.

Currently you are unable to move to any screen except the first one, if internet is disabled and available only through SOCKS5 proxy. This is a really fail.

P.S. Another example of the [request](https://github.com/Aseman-Land/Cutegram/issues/227#issuecomment-232007675).

**Comment by baimafeima** [comment]
Yeah, I can confirm this. Another alternative is to offer obfuscation options such as XOR and obfs4. As of now, it is pretty much unusable for a standard user in above-mentioned countries. I am based in China.

**Comment by ghsrcgh** [comment]
Dear developers, please do something with that. Please move a SOCKS5-configuration menu to the main screen or make a possibility to run application with a special command line to specify a SOCKS5-proxy from real beginning of work.

**Comment by NikolayUvarov** [comment]
Cannot run program without preconfigured proxy.
Now is 2018, April and the worse future had come to Russia.
